File: Sorting Algorithms/insertionSort.js
function insertionSort(arr) {
  if (arr.length <= 1) return arr;

  let result = [arr[0]];
  for (let i = 0; i < arr.length; i++) {
    for (let j = result.length-1; j >= 0; j--) {
      if (arr[i] > result[j]) {
        console.log(result);
        let left = result.splice(0,j+1);
        left.push(arr[i]);
        result = left.concat(result);
        break;
      }
    }
  }
  return result;
}

const arr = [6,5,3,1,8,7,2,4];
console.log(insertionSort(arr));

File: Sorting Algorithms/insertion_sort.rb
def insertion_sort(arr)
  return arr if arr.length <= 1

  result = [arr[0]]

  (1..arr.length-1).each do |idx|
    idx2 = result.length-1
    while idx2 >= 0
      if arr[idx] > result[idx2]
        left = result[0..idx2]
        left.push(arr[idx])
        right = result[idx2+1..-1]
        result = left + right
      end
      idx2 -= 1
    end
  end

  result
end

arr = [6,5,3,1,8,7,2,4];
p insertion_sort(arr);
